hbase-common/src/main/java/org/apache/hadoop/hbase/io/encoding/EncodingState.java

```diff
@@ -32,11 +32,32 @@
    */
   protected Cell prevCell = null;
 
+  // Size of actual data being written. Not considering the block encoding/compression. This
+  // includes the header size also.
+  protected int unencodedDataSizeWritten = 0;
+
+  // Size of actual data being written. considering the block encoding. This
+  // includes the header size also.
+  protected int encodedDataSizeWritten = 0;
```

Comment:
For Row_Index_V1 perspective this is not the encodeddataSize - rather it is metadata. Am not sure how we differentiate it though.